app/components/stacks-by-organization/component.js

```diff
@@ -0,0 +1,24 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  organizationGroups: function() {
+    var groups = Ember.A([]);
+
+    this.get('stacks').forEach(function(stack) {
+      var currentOrganizationId = stack.get('_data.links.organization');
+      var currentOrganization = stack.get('organization');
+
+      if(!groups.findBy('organizationId', currentOrganizationId)) {
+        groups.addObject(Ember.Object.create({
+          organizationId: currentOrganizationId,
+          organization: currentOrganization,
+          stacks: Ember.A([])
+        }));
+      }
+
+      groups.findBy('organizationId', currentOrganizationId).get('stacks').addObject(stack);
+    });
```

Comment:
As written, this logic will always traverse `groups` twice for each stack. It can be made more performance by using a reference to the found organization:

```js
  organizationGroups: function() {
    var groups = [];
    this.get('stacks').forEach(function(stack) {
      var currentOrganizationId = stack.get('_data.links.organization');
      var currentOrganization = stack.get('organization');
      var group = groups.findBy('organizationId',currentOrganizationId);
      if (!group) {
        group = {
          organizationId: currentOrganizationId,
          organization: currentOrganization,
          stacks: []
        };
        groups.push(group);
      }
      group.stacks.push(stack);
    });
    return groups;
  }).property('stacks.[]')
```

This version will only traverse groups once for each stack.

Note that I'm not using Ember's object and array classes. You really only need Ember objects when you want something that is **observable**. In this case, the data structure is only created once then updated. Hence it does not need to use Ember objects.

Because Ember ships with prototype extensions enabled by default (this may change in 2.0), you can still use `findBy` on the native array.

Also note that I've make the property dependent upon `stacks.[]`. The `.[]` means "observe the contents of this array". Without the brackets, only the array itself would be observed, and since arrays are references (the array itself is the same no matter how the contents change), your property would not become invalid and re-compute if new records were added.